File: Data Structures & Algorithms/minimum-recolors-to-get-k-consecutive-black-blocks/submission-1.cpp
class Solution {
public:
    int minimumRecolors(string blocks, int k) {
        int w=0,b=0;
        int l=0,r=0;
        int res=INT_MAX;
        while(r<blocks.length()){
            if(blocks[r]=='W'){
                w++;
            }else{
                b++;
            }

            if(r-l+1<k){
                r++;
                continue;
            }
            res = min(res,w);
            if(blocks[l]=='W'){
                w--;
            }else{
                b--;
            }
            
            r++;
            l++;
        }
        return res;
    }
};

File: Data Structures & Algorithms/non-cyclical-number/submission-1.py
class Solution:
    def isHappy(self, n: int) -> bool:
        m =set()
        while n != 1:
            s=0
            while n:
                s+=(n%10)**2
                n//=10
            if s in m:
                return False
            n=s
            m.add(n)
            
        return True